### PR TITLE
Add guide on things to do when creating new component

### PR DESCRIPTION
**Pull request by ghost**
**Overall change:** _Add documentation of things to do when creating a new component_

**Code changes:**
- Add `Things-to-do-when-creating-a-new-component.md` file with the guide.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by jamesdonoh** [comment]
Looks good, my only suggestion would be 1) rename it `Creating-a-new-component.md` just because the current file name is quite long and 2) link to it from the top-level README.md and also packages/README.md to help people find it.

File: Things-to-do-when-creating-a-new-component.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+| Translations requirements short/long term                                     |                                                          |
+| Split high level requirements into subtasks                                   | During dev swarm                                         |
+| Confirm that the designs are up to date                                       | start                                                    |
+| Does component require event tracking?                                        | start (once event tracking is complete)                  |
+| Decide where this should be developed in pair/individually                    | start                                                    |
+| Discuss unit test implementation within refinement                            | Refinement                                               |
+| Do we extend the swarming approach beyond devs (include design/test too)?     | start                                                    |
+| Dont forget IE and rtl and high contrast mode FF                              |                                                          |
```
Comment (DenisHdz) [inline Things-to-do-when-creating-a-new-component.md]:
```suggestion
| Don't forget about IE, `rtl` and High Contrast mode in FF                              |  all the way through |
```

File: Things-to-do-when-creating-a-new-component.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  - How will this component behave if JavaScript is disabled? Core content needs to be accessible with JavaScript disabled - ensure that this component allows for this.
+  - As a group, break down the work required into small subtasks, and add these to a checklist in the GitHub issue\*. If a subtask requires more effort, members of the swarm can either all work together, or pair on the task. Smaller tasks can be picked up individually. Each of these subtasks can go into code review on their own, rather than waiting for all of the work to be done. You may think of new tasks once you’ve started developing, so feel free to keep adding to this list once you’ve started.
+  - Make sure that accessibility requirements are considered throughout this process, not just at the end during the accessibility swarm.
+  - Some things to consider that are often missed until code review/testing are high contrast mode in firefox, support for right-to-left languages, and IE support.
+
+Do we want to do this, or create a new issue for each subtask?
```
Comment (DenisHdz) [inline Things-to-do-when-creating-a-new-component.md]:
```suggestion

```

File: Things-to-do-when-creating-a-new-component.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| Think about AMP                                                               | start                                                    |
+| Think about non-JS                                                            | start                                                    |
+| Involve testers in early discussions of component                             | start                                                    |
+| Consider accessibility early on                                               | start                                                    |
+| Requirements                                                                  | start                                                    |
+| Acceptance criteria                                                           | start                                                    |
+| Be clear about where requirements and ACs live - Jira/Zeplin/Docs/Dropbox etc | all the way through                                      |
+| Translations requirements short/long term                                     |                                                          |
```
Comment (DenisHdz) [inline Things-to-do-when-creating-a-new-component.md]:
```suggestion
| Translations requirements short/long term                                     |    start/end                                                      |
```

**Comment by ghost** [comment]
@DenisHdz Updated with the fixes.
@jamesdonoh Renamed the file and included it in the readme's

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 2.0.36 | [PR#2826](https://github.com/bbc/psammead/pull/2828) Add quide on creating new components |
```
Comment (oluoluoxenfree) [inline CHANGELOG.md]: Should this be 'add guide'?

File: Creating-a-new-component.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  - If you need to pass in extra props from Simorgh into a Psammead component, but don’t use those props in the components itself, you can spread extra props on the component rather than adding them explicitly. Examples of when we do this are for event tracking and adding ids to elements to be targeted when working with AMP events and state. See an example here: https://github.com/bbc/psammead/blob/latest/packages/components/psammead-brand/src/index.jsx#L200
+  - How will this component behave if JavaScript is disabled? Core content needs to be accessible with JavaScript disabled - ensure that this component allows for this.
+  - As a group, break down the work required into small subtasks, and add these to a checklist in the GitHub issue\*. If a subtask requires more effort, members of the swarm can either all work together, or pair on the task. Smaller tasks can be picked up individually. Each of these subtasks can go into code review on their own, rather than waiting for all of the work to be done. You may think of new tasks once you’ve started developing, so feel free to keep adding to this list once you’ve started.
```
Comment (oluoluoxenfree) [inline Creating-a-new-component.md]: Where does the asterisk here go off to?

File: Creating-a-new-component.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  - Make sure that accessibility requirements are considered throughout this process, not just at the end during the accessibility swarm.
+  - Some things to consider that are often missed until code review/testing are high contrast mode in firefox, support for right-to-left languages, and IE support.
+
+| Thing to remember                                                             | Stage to do the thing at eg start/end, before design etc |
+| ----------------------------------------------------------------------------- | -------------------------------------------------------- |
+| Write e2e tests (Think about e2es)                                            | end? (start)                                             |
```
Comment (oluoluoxenfree) [inline Creating-a-new-component.md]: Maybe put write and think about e2es on different lines? I was a bit confused reading this the first time!

Comment (ghost) [reply Creating-a-new-component.md]: Good call. Done

File: Creating-a-new-component.md (hunk 5 above)
Comment (ghost) [reply Creating-a-new-component.md]: Removed the asterisks.

File: CHANGELOG.md (hunk 4 above)
Comment (ghost) [reply CHANGELOG.md]: Fixed the typo

**Review by oluoluoxenfree — APPROVED** [review]

**Pull request merged**